packages/relay/src/lib/eth.ts

```diff
@@ -970,7 +970,7 @@ export class EthImpl implements Eth {
    * @param call
    * @param blockParam
    */
-  async call(call: any, blockParam: string | null, requestId?: string): Promise<string | JsonRpcError> {
+  async call(call: any, blockParam: string | null, requestId?: string, isFallback?: boolean): Promise<string | JsonRpcError> {
```

Comment:
Hmm not a fan of modifying the method signature just to support an internal retry.
Might be better to pull out the 2 pathways into 2 method e.g. contractCallMirrorNode() & contractCallConsenssusNode()
Then something like
```
if (process.env.ETH_CALL_CONSENSUS == 'false') {
  try {
    ...
    contractCallMirrorNode();
    ...
  } catch (e: any) {
    ...
    if (e instanceof MirrorNodeClientError && e.isNotSupported()) {
        return contractCallConsenssusNode(...);
    }
  }
} else {
  ...
  contractCallConsenssusNode(...);
  ...
}
```

Reply:
I agree on this, to improve code readability and maintenability

Reply:
Agree, I've refactored `eth_call` method.